### PR TITLE
Add a new search section to the help popover

### LINKED ISSUE #1762: explain search syntax (better)
From a user...

> It would also be good to have a list of all the search params you can use in the search field, for example I had to figure out I can use namespace:prod

### PR DESCRIPTION
**Pull request by foot**
Fixes #1762

### CODE REVIEW

**Comment by foot** [comment]
screen shot 2016-10-12 at 17 37 05

**Comment by davkal** [comment]
UI:

> Any field matching pid for the value 12345

I'd highlight `pid`

> memory less than 4mb

Should be 10mb

Generally I'd visually distinguish the input string a bit more from the explanation.

**Comment by davkal** [comment]
The adaptive field help works nicely!

File: client/app/scripts/components/help-panel.js [hunk 1]
```diff
@@ -27,20 +35,135 @@ function renderShortcuts(cuts) {
...
+
+const REGEX_SEARCHES = [
+  {term: 'foo|bar', label: 'All fields for foo or bar'},
+  {term: 'command: foo(bar|baz)', label: 'Command field for foobar or foobaz'},
+];
+
+
+const METRIC_SEARCHES = [
+  {term: 'cpu > 4%', label: 'CPU greater than 4%'},
+  {term: 'memory < 10mb', label: 'memory less than 4mb'},
```
Comment (davkal) [inline client/app/scripts/components/help-panel.js]: this should be `...than 10MB`, capital M is for Mega, and capital B is for byte, as opposed to lowercase b (bit).

File: client/app/scripts/components/search.js [hunk 2]
```diff
@@ -130,14 +133,17 @@ class Search extends React.Component {
             </label>
           </div>
           {!showPinnedSearches && <div className="search-hint">
-            {getHint(nodes)}
+            {getHint(nodes)} <span className="search-help-link" onClick={onClickHelp}>
```
Comment (davkal) [inline client/app/scripts/components/search.js]: HELP! is hard to read. Why not use a question mark in a circle?

Also, the search bar collapses again when help is clicked, which was a bit confusing.

File: client/app/scripts/selectors/chartSelectors.js [hunk 3]
```diff
@@ -95,6 +96,12 @@ export const dataNodesSelector = createSelector(
 );
 
 
+export const searchableFieldsSelector = createSelector(
```
Comment (davkal) [inline client/app/scripts/selectors/chartSelectors.js]: Shouldnt this be a selector that checks for deep equality (fields are always new sets)?

File: client/app/scripts/utils/string-utils.js [hunk 4]
```diff
@@ -65,7 +65,7 @@ export const formatMetric = makeFormatMetric(renderHtml);
 export const formatMetricSvg = makeFormatMetric(renderSvg);
 export const formatDate = d3.time.format.iso;
 
-const CLEAN_LABEL_REGEX = /\W/g;
+const CLEAN_LABEL_REGEX = /[^A-Za-z]/g;
```
Comment (davkal) [inline client/app/scripts/utils/string-utils.js]: I'm curious, why was that needed?
